test/powershell/Language/Operators/SplitOperator.Tests.ps1

```diff
@@ -30,13 +30,18 @@ Describe "Split Operator" -Tags CI {
             $res[1] | Should Be "b"
             $res[2] | Should Be "c"
             $res[3] | Should Be "d"
+        }
+
+        It "Binary split operator works with negative substring limit" {
+            $res = "a b c d" -split " ", -3
+            $res.count | Should Be 3
+            $res[0] | Should Be "a b"
+            $res[1] | Should Be "c"
+            $res[2] | Should Be "d"
 
             $res = "a b c d" -split " ", -1
-            $res.count | Should Be 4
-            $res[0] | Should Be "a"
-            $res[1] | Should Be "b"
-            $res[2] | Should Be "c"
-            $res[3] | Should Be "d"
+            $res.count | Should Be 1
+            $res[0] | Should Be "a b c d"
```

Comment:
I believe we want to get "d" as a last element.
/cc @mklement0

Reply:
While that might seem intuitive, it would make the `Max-substrings` operand asymmetrical. 

The guiding principal behind the change is that the sign (`+/-`) signifies only the direction of the split operation, whereas the absolute value determines the maximum resulting number of substrings, with 0 preserving the meaning of "unlimited"

Reply:
Actually, I wasn't thinking of _direction_, but _what end_ to apply the split to - elements returned should always reflect the input order.

In this case, due to `-1`, the _last 1_ element should be returned, preceded by the remaining _prefix_, so we should get:

```none
$rest, $result = 'a b c d' -split '', -1
$rest | Should Be 'a b c'
$result.Count | Should be 1
$result[0] | Should Be 'd'
```

Reply:
Whether you call it *direction* or not doesn't change the fact that by splitting from the *end* rather than the *start* we are in fact modifying the direction of the operation :-) 

Wouldn't your proposed result be confusing given the name of the operand (`Max-substrings` rather than `Number-Of-Times-To-Split` or similar)?

Reply:
_Direction_ is ambiguous, as that could be interpreted as wanting to also reverse the order in which the resulting elements are returned (e.g., `'a b c d' -split ' ', -2` resulting in `'a b', 'd, 'c'`, which we don't want), so I wanted to clarify.

To me, `Max-Substrings` makes sense: a negative value means: return `Max-Substrings` elements _from the end_ (in input order).

Reply:
Re `Max-Substrings`: I see your point now. With my proposal, you actually get `Max-SubStrings + 1` elements back - this is the asymmetry you mention.

Reply:
I agree that your approach is better for consistency with how `-split` currently works.

Given that what was approved was my approach, however, we should get approval again: cc @SteveL-MSFT

Reply:
If you think about how negative array indices are treated in PowerShell, it makes perfect sense, and as https://github.com/PowerShell/PowerShell/pull/5125/commits/1d59be0f17e3b4e4b6c5d57daaf882b364f2939f shows, shifting to your original suggestion is trivial, but I would like to keep it symmetric (again, given that the second operand is not an index but a boundary value)

Reply:
Understood and agreed.

I indeed originally had array-index logic in mind, but neglected to consider that the current `-split` behavior uses the-number-of-elements-I-want-plus-one-for-the-rest logic.

Just to spell it out: with your approach (which we should go with), both `1` and `-1` are no-ops - both request no splitting at all, and which end to apply it to is then moot.

Reply:
I agree that the symmetric behavior makes sense and updating the documentation with good examples will clear up any confusion. cc @lzybkr

Reply:
@mklement0 Could you please modify your Issue description to satisfy the behavior?

Reply:
@iSazonov: Thanks for reminding me - done.